### DISCUSSION TITLE
Compression

### DISCUSSION THREAD

**Discussion by JunkYardRobotBoy** [iimachines/Maya2glTF#182] [Ideas]
Apologies if I missed this somewhere in the docs.
Are there any plans to add in Draco compression on export at some stage?
Also is there a way to disable baking the animation per frame in order to create a smaller file size?
This tool is amazing and really appreciate all the work on it!

**Comment by ziriax** [comment]
I started an experiment with curve compression a long time ago, but this was never released, as my customer didn't need it anymore, they used procedural IK animation in the end.  It was a lot of fun, but not easy, it would require commercial support to get this done. Also, it was not compatible with GLTF itself, it used Chebyshev polynomials, but I would have liked to experiment with other techniques too, like Radial Basis Functions, wavelets, arithmetic coding, etc...

Another library from a professional game tool developer exists that tries to do that: https://github.com/nfrechette/acl-gltf

Note that in general you can't export Maya curves as is to GLTF, since GLTF just doesn't support these curves. Maya's (and almost all CG apps) uses curves that are actually 2D freeform curves, not the simply cubic polynomials that GLTF supports. The GLTF 2.0 spec mentions this:

> Animations with non-linear time inputs, such as time warps in Autodesk 3ds Max or Maya, are not directly representable with glTF animations. glTF is a runtime format and non-linear time inputs are expensive to compute at runtime. Exporter implementations should sample a non-linear time animation into linear inputs and outputs for an accurate representation.

You really wouldn't want to do animation with simple cubic polynomials, that would be horrible for an artist I guess, as you can only control the slope of the curve, not a tangent vector, as you are used to.

Furthermore if you have IK or other constraints, then again, you don't end up with simple curves...

No, Draco compression is not supported yet. You can easily apply that as a standalone tool no? A lot of other tools exist in the GLTF pipeline too.
